### ISSUE TITLE
unwrap_used does not detect unwrap/unwrap_err when using fully qualified syntax

### ISSUE THREAD

**Issue by calum4** [rust-lang/rust-clippy#16484]
### Summary

I'd find it staggering if this hasn't been noticed before, so if this is a duplicate issue, apologies. (I couldn't find one)

The lint description does seem to indicate it only detects `.unwrap()`/`.unwrap_err()`, although I'd argue my assumption that this would work is a fair one to make.
> Checks for .unwrap() or .unwrap_err() calls on Results and .unwrap() call on Options.

I'd like to **try** tackle this myself, but won't have the time for likely a couple week so if someone beats me to it, fair enough.

Thanks!

### Lint Name

unwrap_used

### Reproducer

I tried this code:

```rust
#![deny(clippy::unwrap_used)]

fn main() {
    let option_some = Some(());
    assert_eq!(Option::unwrap(option_some), ());

    #[derive(Debug, PartialEq)]
    struct Error;

    let result_ok: Result<(), Error> = Ok(());
    assert_eq!(Result::unwrap(result_ok), ());

    let result_err: Result<(), Error> = Err(Error);
    assert_eq!(Result::unwrap_err(result_err), Error);
}
```

I expected to see this happen:
The lint to complain about the usage of unwrap/unwrap_err

Instead, this happened:
The lint did not complain

### Version

```text
rustc 1.93.0 (254b59607 2026-01-19)
binary: rustc
commit-hash: 254b59607d4417e9dffbc307138ae5c86280fe4c
commit-date: 2026-01-19
host: x86_64-unknown-linux-gnu
release: 1.93.0
LLVM version: 21.1.8
```
